sdk/security_keyvault/examples/get_secret_versions.rs

```diff
@@ -1,21 +1,16 @@
-use azure_identity::DefaultAzureCredentialBuilder;
 use azure_security_keyvault::prelude::*;
 use futures::StreamExt;
-use std::{env, sync::Arc};
+use std::env;
 
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let keyvault_url =
         env::var("KEYVAULT_URL").expect("Missing KEYVAULT_URL environment variable.");
     let secret_name = env::var("SECRET_NAME").expect("Missing SECRET_NAME environment variable.");
 
-    let creds = Arc::new(
-        DefaultAzureCredentialBuilder::new()
-            .exclude_managed_identity_credential()
```

Comment:
This is an all-to-common pattern to avoid the delay cost the managed identity for a VM. By defaulting it to off, the cost is avoided. This aligns with the Rust performance theme.

Those deploying to virtual machines can do:
```
DefaultAzureCredentialBuilder::new()
    .include_virtual_machine_managed_identity_credential()
    .build()?;
```

Or they can set `AZURE_CREDENTIAL_TYPE` to `virtualmachine` and use `SpecificAzureCrednetial` via `azure_identity::create_specific_credential()?` or `azure_identity::create_credential()?`.

Reply:
The DAC is less about speed and more about getting a user-friendly behavior by default. I appreciate the initial (as @demoray pointed out above) delay is less than optimal, but I think having different behavior with DAC is worse. If the delay happened every time, I'm right there with you. But it shouldn't.

Reply:
As an example, a CLI app using DAC that is relying on Azure CLI authentication would hit it every time. Basically, every example we have.